Clean additional classes for sources passed to a compiler (#21878)
gradle/gradle: subprojects/docs/src/docs/userguide/jvm/java_plugin.adoc

[[sec:incremental_compilation_known_issues]]
=== Known issues

* If you are using an annotation processor that reads resources (e.g. a configuration file), you need to declare those resources as an input of the compile task.
* If a resource file is changed, Gradle will trigger a full recompilation.

> * If a custom `executable` or `javaHome` is configured on the compile task some optimizations will be deactivated, e.g. compile task will not be incremental after a compile error or if a Java constant is changed. It's recommended to use <<toolchains.adoc#toolchains, toolchains>> instead if possible.

Reviewer: ```suggestion
* Using a custom `executable` or `javaHome` deactivates some optimizations. The compile task does not use incremental build immediately after a compile error or if a Java constant changes. Use <<toolchains.adoc#toolchains, toolchains>> instead if possible.
```
Suggestion: slight rephrase.

Author: Applied suggestions
